### PR TITLE
Added CONTRIBUTING.md and modified README.md

### PR DESCRIPTION
**Pull request by santacodes**
I used the [PyBaMM's CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md) file as a template for this project as well. It was modified and omitted for it to be in accordance with this project. `README.md` file had to be written mostly on its own.

@brosaplanella sorry for the delay but I hope it helps with the setup process for your workshop :).  The parameter sets have to be included within the template in order for them to get copied to the user-generated cookiecutter project, if you want to use the parameter sets feature I suggest you do an editable install to the `pybamm-cookiecutter` project under a virtual environment, the instructions are in `README.md`.

### CODE REVIEW

**Review by brosaplanella — APPROVED** [review]
Looks good, thanks! Will try to install today :)

**Comment by brosaplanella** [comment]
Initialising the cookiecutter worked well, but can't install using pip (see error below) nor nox because the dev session is not defined.
```bash
Defaulting to user installation because normal site-packages is not writeable
Obtaining file:///home/brosaplanella/chocomaths
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Installing backend dependencies ... done
  Preparing editable metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [30 lines of output]
      Traceback (most recent call last):
        File "/home/brosaplanella/.local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/brosaplanella/.local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/brosaplanella/.local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 176, in prepare_metadata_for_build_editable
          whl_basename = build_hook(metadata_directory, config_settings)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/build.py", line 83, in build_editable
          return os.path.basename(next(builder.build(directory=wheel_directory, versions=['editable'])))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/builders/plugin/interface.py", line 90, in build
          self.metadata.validate_fields()
        File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/metadata/core.py", line 265, in validate_fields
          _ = self.version
              ^^^^^^^^^^^^
        File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/metadata/core.py", line 149, in version
          self._version = self._get_version()
                          ^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/metadata/core.py", line 248, in _get_version
          version = self.hatch.version.cached
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/metadata/core.py", line 1466, in cached
          raise type(e)(message) from None
      LookupError: Error getting the version from source `vcs`: setuptools-scm was unable to detect version for /home/brosaplanella/chocomaths.

      Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

      For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

UPDATE: Actually, the issue was the I didn't initialise git, that's why it broke. We should add a line telling people to initialise git before installing.

**Comment by santacodes** [comment]
> Initialising the cookiecutter worked well, but can't install using pip (see error below) nor nox because the dev session is not defined.
> 
> ```shell
> Defaulting to user installation because normal site-packages is not writeable
> Obtaining file:///home/brosaplanella/chocomaths
>   Installing build dependencies ... done
>   Checking if build backend supports build_editable ... done
>   Getting requirements to build editable ... done
>   Installing backend dependencies ... done
>   Preparing editable metadata (pyproject.toml) ... error
>   error: subprocess-exited-with-error
> 
>   × Preparing editable metadata (pyproject.toml) did not run successfully.
>   │ exit code: 1
>   ╰─> [30 lines of output]
>       Traceback (most recent call last):
>         File "/home/brosaplanella/.local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
>           main()
>         File "/home/brosaplanella/.local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
>           json_out['return_val'] = hook(**hook_input['kwargs'])
>                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "/home/brosaplanella/.local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 176, in prepare_metadata_for_build_editable
>           whl_basename = build_hook(metadata_directory, config_settings)
>                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/build.py", line 83, in build_editable
>           return os.path.basename(next(builder.build(directory=wheel_directory, versions=['editable'])))
>                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/builders/plugin/interface.py", line 90, in build
>           self.metadata.validate_fields()
>         File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/metadata/core.py", line 265, in validate_fields
>           _ = self.version
>               ^^^^^^^^^^^^
>         File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/metadata/core.py", line 149, in version
>           self._version = self._get_version()
>                           ^^^^^^^^^^^^^^^^^^^
>         File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/metadata/core.py", line 248, in _get_version
>           version = self.hatch.version.cached
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "/tmp/pip-build-env-nrjma6ib/overlay/local/lib/python3.11/dist-packages/hatchling/metadata/core.py", line 1466, in cached
>           raise type(e)(message) from None
>       LookupError: Error getting the version from source `vcs`: setuptools-scm was unable to detect version for /home/brosaplanella/chocomaths.
> 
>       Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.
> 
>       For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: metadata-generation-failed
> 
> × Encountered error while generating package metadata.
> ╰─> See above for output.
> 
> note: This is an issue with the package mentioned above, not pip.
> hint: See above for details.
> ```
> 
> UPDATE: Actually, the issue was the I didn't initialise git, that's why it broke. We should add a line telling people to initialise git before installing.

I added it, but I think part of that problem also has to do with the absence of `nox dev` session in this particular PR branch. It only exists in the other `parametersets` branch in case you tried cookiecutter with this or the `main` branch. Also a git repository should be automatically initialised if a user uses this template to make their project within GitHub itself. I am not sure about pip though, if it would initialise a git repository or not. Maybe we could add a prompt and a script to ask the user `Initialise git?` and if it's a yes we execute `git init` through the script, or we just make it mandatory for all to have the project as a git repository.

**Comment by brosaplanella** [comment]
I cloned the repo and initialised it locally (only pushed to Github later) so had to initalise git manually. I suspect if you initialise directly on Github then you just clone the remote repo locally which already creates a `.git` folder.

**Review by agriyakhetarpal — APPROVED** [review]
I think this looks good to me for the initial go. We will iterate on this in further passes before we push the code to PyPI, and I see no reason to be too pedantic :) Thanks, @santacodes!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+### Building the documentation
+
+To test and debug the documentation, it's best to build it locally. To do this, navigate to your pybamm-cookiecutter directory in a console, and then type (on GNU/Linux, macOS, and Windows):
+
+```
+nox -s docs
+```
+
+And then visit the webpage served at `http://127.0.0.1:8000`. Each time a change to the documentation source is detected, the HTML is rebuilt and the browser automatically reloaded. In CI, the docs are built and tested using the `docs` session in the `noxfile.py` file with warnings turned into errors, to fail the build. The warnings can be removed or ignored by adding the appropriate warning identifier to the `suppress_warnings` list in `docs/conf.py`.
```
[the remark this answers is not in the document]
Comment (arjxn-py) [reply CONTRIBUTING.md:118]: further tracked in #20

File: README.md [hunk 2]
```diff
@@ -22,5 +22,59 @@
 [pypi-version]:             https://img.shields.io/pypi/v/pybamm-cookiecutter
 [rtd-badge]:                https://readthedocs.org/projects/pybamm-cookiecutter/badge/?version=latest
 [rtd-link]:                 https://pybamm-cookiecutter.readthedocs.io/en/latest/?badge=latest -->
+[![Powered by NumFOCUS](https://img.shields.io/badge/powered%20by-NumFOCUS-orange.svg?style=flat&colorA=E1523D&colorB=007D8A)](http://numfocus.org)
```
[the remark this answers is not in the document]
Comment (arjxn-py) [reply README.md:25]: Other badges are not being rendered because they're not ready to show yet, we can move the text of other badges once they're ready.

**Review by arjxn-py — APPROVED** [review]
Thanks @santacodes, i think this should be good to merge now. Knowing that we'd be coming back to this in other PR's as well as things are progressing.

File: README.md [hunk 3]
```diff
@@ -22,5 +22,59 @@
...
 
 This repository contains a `cookiecutter` template for battery modeling projects using PyBaMM, released under the [BSD-3-Clause license](LICENSE). Currently under active development.
+
+## 📄 Using pybamm-cookiecutter
+
+### Generating projects with pybamm-cookiecutter
```
[the remark this answers is not in the document]
Comment (arjxn-py) [reply README.md]: Marking this as resolved as I think that this suggestion been taken care of.

**Pull request merged**